graql/reasoner/atom/binary/RelationAtom.java

```diff
@@ -186,10 +190,22 @@ public IsaAtom toIsaAtom() {
     }
 
     @Override
-    public Set<Atom> rewriteToAtoms() {
-        return this.getRelationPlayers().stream()
-                .map(rp -> create(relationPattern(getVarName().asReturnedVar(), Sets.newHashSet(rp)), getPredicateVariable(), getTypeLabel(), null, getParentQuery(), context()))
-                .collect(Collectors.toSet());
+    public Stream<Atom> rewriteToAtoms() {
+        // we keep duplicate role players together as part of a single atom, as the alternative,
```

Comment:
I'm wondering if this rewrite is correct now in general:

Q: 
$r (anotherRole: $x, anotherRole: $y) 

would get rewritten to:
Q': 
```
$r (anotherRole: $x);
$r (anotherRole: $y);
```

If we allow the edges to repeat, Q explicitly requires 2 RP edges, Q' requires one edge which might be double.
Example:
For a single relation in the DB with RP configuration: `(anotherRole: A, anotherRole: B)`
Q returns: 
[$x/A, $y/B], 
[$x/B, $y/A]

whereas Q' returns:
[$x/A, $y/B]
[$x/B, $y/A]
[$x/A, $y/A]
[$x/B, $y/B]